crates/milli/src/filterable_attributes_rules.rs

```diff
@@ -0,0 +1,204 @@
+use deserr::{DeserializeError, Deserr, ValuePointerRef};
+use serde::{Deserialize, Serialize};
+use std::collections::{BTreeSet, HashSet};
+use utoipa::ToSchema;
+
+use crate::{
+    attribute_patterns::{match_distinct_field, match_field_legacy, PatternMatch},
+    constants::RESERVED_GEO_FIELD_NAME,
+    AttributePatterns, FieldsIdsMap,
+};
+
+#[derive(Serialize, Deserialize, PartialEq, Eq, Clone, Debug, ToSchema)]
+#[serde(untagged)]
+pub enum FilterableAttributesRule {
+    Field(String),
+    Pattern(FilterableAttributesPatterns),
+}
+
+impl FilterableAttributesRule {
+    pub fn match_str(&self, field: &str) -> PatternMatch {
+        match self {
+            FilterableAttributesRule::Field(pattern) => match_field_legacy(pattern, field),
+            FilterableAttributesRule::Pattern(patterns) => patterns.match_str(field),
+        }
+    }
+
+    pub fn has_geo(&self) -> bool {
+        matches!(self, FilterableAttributesRule::Field(field_name) if field_name == RESERVED_GEO_FIELD_NAME)
+    }
+
+    pub fn features(&self) -> FilterableAttributesFeatures {
+        match self {
+            FilterableAttributesRule::Field(_) => FilterableAttributesFeatures::legacy_default(),
+            FilterableAttributesRule::Pattern(patterns) => patterns.features(),
+        }
+    }
+}
+
+#[derive(Serialize, Deserialize, PartialEq, Eq, Clone, Debug, Deserr, ToSchema)]
+#[serde(deny_unknown_fields, rename_all = "camelCase")]
+#[deserr(rename_all = camelCase, deny_unknown_fields)]
+pub struct FilterableAttributesPatterns {
+    pub patterns: AttributePatterns,
+    #[serde(default)]
+    #[deserr(default)]
+    pub features: FilterableAttributesFeatures,
```

Comment:
the usage pages sometimes states that this should be `filter`. Am I misunderstanding something?

Reply:
the `filter` is contained in the `features` with the `facet_search`